core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/DataframeConvertable.kt

```diff
@@ -0,0 +1,11 @@
+package org.jetbrains.kotlinx.dataframe.api
+
+import org.jetbrains.kotlinx.dataframe.DataFrame
+
+/**
+ * Common interface for all objects that can be converted to [DataFrame].
+ * Useful for rendering in Jupyter integration.
+ */
+public interface DataframeConvertable {
```

Comment:
Since it's useful mostly for Jupyter Integration, let's make in an internal interface? I imagine having "df.update("").convertToDataFrame()" in completion might be somewhat confusing

Reply:
Yes, you are right. It's unacceptable to pollute high-level API. But the internal interface will be hard to maintain because, in the future, it will require not to forget to implement DataframeConvertable on every new implementation of core interfaces such as DataColumn, GroupBy, etc. And it will be hard to understand why rendering is not working as expected.
What do you think if we eliminate the idea of a common interface for conversion and just make a conversion function?
````
public fun convertToDataFrame(dataframeLike: Any): DataFrame<*> =
    when (dataframeLike) {
        is Pivot<*> -> dataframeLike.frames().toDataFrame()
        is ReducedPivot<*> -> dataframeLike.values().toDataFrame()
        is PivotGroupBy<*> -> dataframeLike .frames()
        is ReducedPivotGroupBy<*> -> dataframeLike.values()
        is SplitWithTransform<*, *, *> -> dataframeLike.into()
        is Merge<*, *, *> -> dataframeLike.into("merged")
        is Gather<*, *, *, *> -> dataframeLike.into("key", "value")
        is Update<*, *> -> dataframeLike.df
        is Convert<*, *> -> dataframeLike.df
        is FormattedFrame<*> -> dataframeLike.df
        is AnyCol -> dataFrameOf(dataframeLike)
        is AnyRow -> dataframeLike.toDataFrame()
        is GroupBy<*, *> -> dataframeLike.toDataFrame()
        is AnyFrame -> dataframeLike
        else -> throw IllegalArgumentException("Unsupported type: ${dataframeLike::class}")
    }
`
At least it will fail soon at runtime with a clear exception.

Reply:
Sure, sounds good